other/upgrade_2-1_mysql.sql

```diff
@@ -115,6 +115,43 @@ if (@$modSettings['smfVersion'] < '2.1')
 ---}
 ---#
 
+---# Parsing board descriptions and names
+---{
+if (@$modSettings['smfVersion'] < '2.1')
```

Comment:
Please had in mind that we got rc1 to rcX support, this would break here

Reply:
The parsing should be done just once. On that same file the same version compare [was used](https://github.com/SimpleMachines/SMF2.1/blob/2a6d524410429ae4cf11fb8f1bf7b65d67f100e2/other/upgrade_2-1_mysql.sql#L96) and merged into release-2.1 branch, hence why I use it.

Reply:
Yes,
this exists before of realsing rc1,
you change came after this thow it's to late to use this kind of checks,
because already working 2.1 setup will not run this in upgrade path which they need.

Reply:
What do you recommend then?

Reply:
What would be happen when you run the code twice?
Which logic get here the problem?

Reply:
It should run only once, when a user goes from 2.0 to 2.1, if the same user for whatever reason does another upgrade from 2.1 to 2.1 this particular step should not be executed.

Reply:
I ask you why?
Which logic get broken here?

Reply:
I don't really understand what you mean, if you ask why this has to be done once, here's why:

2.0 supports adding raw HTML to board descriptions and I think categories too, can't remember.

2.1 doesn't allow this anymore, in fact, it doesn't allow to add BBC either expect by the ones specified in $context['description_allowed_tags'].

In order to provide a better upgrade experience for those users who had HTML on their description boards, an attempt to parse their HTML into BBC is done. Any description is now saved in the DB as BBC and that BBC gets parsed on display only, this is why this step has to be done just once.

Reply:
So it's create no issue when it run twice?

Reply:
It shouldn't but in practice things can and surely will get broken since already htmlspecialchars HTML will get another round of htmlspecialchars, making it look quite broken for the user.

But more importantly, why run things twice where there isn't any need for? the upgrade isn't service friendly so if we can avoid doing unneeded steps then lets do it.

Reply:
Don't let's do thing just we can,
you need to concider that you had to choose between run once and never or once and multiple time.

So instead of hurting yourself,
you could consider if it possible to write the code in a way how is able to run multiple times without hurting.

When it's very important to you to run this realy once,
than write in the setting or theme table a row that this task is done and
check on a rerun if the task is marked as done.

Reply:
???

I did consider the options and the best one is to do this step only when they upgrade from 2.0 to 2.1.

The same DB will only do this upgrade once (2.0 to 2.1), they can of course do another upgrade, in that case it will be a 2.1 to 21. upgrade on which case this step is useless and we shouldn't waste CPU cicles on it.

I really don't think a setting or a theme setting is required here, $modSettings['smfVersion'] < '2.1' fulfills our needs quite well, we only need to run this when a user goes from 2.0 to 2.1, that's it.

From what I can see, the $modSettings['smfVersion'] < '2.1' check won't pass for any RC or beta version and that's OK, in fact it is exactly what we want, we only want this piece of code to be executed when there's a 2.0 to 2.1 upgrade, all other scenarios don't matter to us.

Reply:
I find this interesting that more important for you,
that the upgrade process runs fast,
instead of reliable.

solution already exists how could fix it: https://www.simplemachines.org/community/index.php?topic=548174.0
but was decided by persons not to take this way.

To get to the point,
it can happen in the upgrade process that error apears which can let you upgrade process skip somewhere in the middle but the mod version get raised up,
when the admin start again the upgrade you your partial or never hittet upgrade skript part will be never vistid again --> weak reliability

Reply:
You do realize that this issue has nothing to do with partial upgrades?

If you want to yet again give it a shot and discuss partial upgrades then open a separate issue or bump that SMF topic.

Partial upgrades from beta or RC for this particular case? not important at all, we only care about 2.0 to 2.1.

If the upgrade process fails then all the user has to do is go to admin/boards/edit board and save the page, that's it. Saving the edits will trigger the exact same piece of code the upgrade failed to do.

This isn't a "must have" or "must execute" thing, its merely a way to offer a better upgrade process, it isn't a required step and shouldn't be any issues if the upgrade fails to execute this step as its quite simple to manually make the change if needed.